### PR TITLE
Cleaning the repo from old code

### PR DESCRIPTION
**Pull request by jstypka**
- remove all the experimental code used for trying to figure out how Magpie should work a long time ago. We're now down to <1000 lines altogether! 🎉 
 - move the main API code from `__init__.py` to `main.py`
 - make Travis scan the whole project and not only the test directory (now it's gonna fail the build e.g. if some imports don't make sense)

@kaplun

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,4 +27,4 @@ install:
   - python setup.py install
 
 script:
-  - python -m unittest discover --start-directory magpie/tests
+  - python -m unittest discover
```
Comment (kaplun) [inline .travis.yml]: On INSPIRE we now use the much more powerful `py.test` and tests are kept outside of the code base. Maybe this is some further improvement (not urgent) to make the codebase even more clean.

Comment (jstypka) [inline .travis.yml]: Agreed. So far though there's just one test, so I'm not sure if it's worth to introduce a new framework for that :-)

File: magpie/utils.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 except ImportError:
```
Comment (kaplun) [inline magpie/utils.py]: For me in order to review this PR, can you explain better what you did in this commit log? What were these `forgotten functions` and in which sense you have updated them?

File: magpie/__init__.py [hunk 3]
```diff
@@ -1,273 +1 @@
-from __future__ import unicode_literals, print_function, division
```
Comment (kaplun) [inline magpie/__init__.py]: Why moving content of `__init__.py` into `main.py`? I am not sure there is a real benefit :)

File: .travis.yml (hunk 1 above)
Comment (kaplun) [inline .travis.yml]: Eheh, we should add tests then :+1:

File: magpie/__init__.py (hunk 3 above)
Comment (jstypka) [inline magpie/__init__.py]: You know Python better than me, but as far as I've seen `__init__.py` files are almost always empty or just having imports while the real code is split into appropriately named files.

I don't care too much to be honest, this seemed a little cleaner to me :)

File: .travis.yml (hunk 1 above)
Comment (jstypka) [inline .travis.yml]: sure, but that would probably go in another PR :)

File: magpie/utils.py (hunk 2 above)
Comment (jstypka) [inline magpie/utils.py]: Ah yeah, I forgot to describe that. So in `magpie/utils.py` there are a couple of functions that are not directly called anywhere, but might be useful for people using the software for analysis of their corpus.

One is for computing the frequency of classes in a corpus other gives you top N classes in the corpus and things like that. I decided not to remove them, because they might come in handy for somebody, so I just updated them and their docs.
